src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let stateCopy = Object.assign({}, state);
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        stateCopy = Object.assign({}, stateCopy, key.extraData);
+        break;
+
+      case 'removeProperties':
+        stateCopy = Object.assign({}, stateCopy);
```

Comment:
[DONT REPEAT YOURSELF]: Since you are creating a new object copy in every case of the switch statement, you can move this operation outside of the switch block to avoid repetition.